src/tdastro/example_runs/simulate_snia.py

```diff
@@ -16,48 +19,173 @@
+        passband_list.append({"filter_name": band, "table_path": file_path})
+        logger.info(f"Loading band {band} from {file_path}")
+
+    if len(passband_list) == 0:
+        raise ValueError("No passbands being loaded.")
+
+    # Do the actual loading and processing.
+    passbands = PassbandGroup(
+        passband_parameters=passband_list,
+        survey="LSST",
+        units="nm",
+        trim_quantile=0.001,
+        delta_wave=1,
+    )
+
+    # Register sncosmo bandpasses
+    for f, passband in passbands.passbands.items():
+        sncosmo_bandpass = sncosmo.Bandpass(*passband.processed_transmission_table.T, name=f"tdastro_{f}")
+        sncosmo.register(sncosmo_bandpass, force=True)
+
+    return passbands
+
 
 def draw_single_random_sn(
     source,
     opsim,
     passbands,
+    state=None,
 ):
+    """Process a single random SN realization.
+
+    Parameters
+    ----------
+    source : PhysicalModel
+        The PhysicalModel to use for the flux computation.
+    opsim : OpSim
+        The OpSim for the simulations
+    passbands : PassbandGroup
+        The passbands to use in generating the observations.
+    state : GraphState
+        The sample values to use. If None resamples the state.
+
+    Returns
+    -------
+    res : dict
+        A dictionary of useful information about the run.
     """
-    Draw a single random SN realiztion
-    """
+    if state is None:
+        state = source.sample_parameters()
 
-    state = source.sample_parameters()
+    # Extract some important parameters that we need to use.
+    ra = state["source"]["ra"]
+    dec = state["source"]["dec"]
+    t0 = state["source"]["t0"]
+    z = state["source"]["redshift"]
 
-    z = source.get_param(state, "redshift")
+    # Compute the rest wavelength information and save it.
     wave_obs = passbands.waves
     wavelengths_rest = wave_obs / (1.0 + z)
-
     res = {"wavelengths_rest": wavelengths_rest}
 
-    t0 = source.get_param(state, "t0")
-
-    ra = source.get_param(state, "ra")
-    dec = source.get_param(state, "dec")
+    # Find the times at which this source is seen.
     obs_index = np.array(opsim.range_search(ra, dec, radius=1.75))
 
     # Update obs_index to only include observations within SN lifespan
     phase_obs = opsim["time"][obs_index] - t0
     obs_index = obs_index[(phase_obs > -20 * (1.0 + z)) & (phase_obs < 50 * (1.0 + z))]
 
+    # Extract the timing and filter information for those observations, changing the
+    # match band names in passbands object.
     times = opsim["time"][obs_index].to_numpy()
-    filters = opsim["filter"][obs_index].to_numpy(str)
-    # Change to match band names in passbands object
-    filters = np.char.add("LSST_", filters)
-
     if len(times) == 0:
-        print(f"No overlap time in opsim for (ra,dec)=({ra:.2f},{dec:.2f})")
-        return
-
+        raise ValueError(f"No overlap time in opsim for (ra,dec)=({ra:.2f},{dec:.2f})")
```

Comment:
I don't think we want error here because we want to keep the loop running.

Reply:
I made it a warning. Let me know if you'd like me to move it back to a print statement or debug logging or something else.

Reply:
Warning should work for now. Thanks!